caikit/core/model_manager.py

```diff
@@ -67,12 +69,119 @@ class ModelManager:
+                train function
+
+        Kwargs:
+            trainer (Union[str, ModelTrainerBase]): The trainer to use. If given
+                as a string, this is a key in the global config at
+                model_management.trainers.
+            save_path (Optional[str]): Path on disk where the model should be
+                saved (may be relative to a remote trainer's filesystem)
+            save_with_id (bool): Inject the training ID into the save path for
+                the output model
+            wait (bool): Wait for training to complete before returning
+            **kwargs: Additional keyword arguments to pass through to the
+                modules's train function
+
+        Returns:
+            model_future (ModelTrainerBase.ModelFutureBase): The future handle
+                to the model which holds the status of the in-flight training.
+        """
+        # Resolve the module class
+        if isinstance(module, str):
+            module_id = module
+            module = module_registry().get(module_id)
+            error.value_check(
+                "<COR00469102E>",
+                module is not None,
+                "Unable to train unknown module {}",
+                module_id,
+            )
+        error.subclass_check("<COR05418775E>", module, ModuleBase)
+
+        # Get the trainer to use
+        trainer = self._get_trainer(trainer)
+
+        # Start the training
+        with alog.ContextTimer(log.debug, "Started training in: "):
+            model_future = trainer.train(
+                module,
+                *args,
+                save_path=save_path,
+                save_with_id=save_with_id,
+                **kwargs,
+            )
+            log.debug(
+                "Started training %s with save path %s",
+                model_future.id,
+                model_future.save_path,
+            )
+
+        # If requested, wait for the future to complete
+        if wait:
+            log.debug("Waiting for training %s to complete", model_future.id)
+            with alog.ContextTimer(
+                log.debug, "Finished training %s in: ", model_future.id
+            ):
+                model_future.wait()
+
+        # Return a handle to the training
+        return model_future
+
+    def get_model_future(
+        self,
+        training_id: str,
+    ) -> ModelTrainerBase.ModelFutureBase:
+        """Get the future handle to an in-progress training
+
+        Args:
+            training_id (str): The ID string from the original training
+                submission's ModelFuture
+
+        Returns:
+            model_future (ModelTrainerBase.ModelFutureBase): The future handle
+                to the model which holds the status of the in-flight training.
+        """
+        try:
+            trainer = self._get_trainer(ModelTrainerBase.get_trainer_name(training_id))
+
+        # Fall back to the default trainer to try to find this ID
```

Comment:
This is done to support external IDs when running locally for Model Train. These external IDs will not contain the trainer name prefix since they'll come from an external system